### DISCUSSION TITLE
MP problems (English version)

### DISCUSSION THREAD

**Discussion by c0nnex** [Mogli12/KeyboardSteer#752] [Q&A]
- VCA HUD is not displaying
- Settings should be saved per Player per Vehicle (imho). Right now it seems to be saved only per vehicle. I play with keyboard, buddy with wheel. he uses completely diffrent steering/gas/brake settings
- Sometimes the border markers of the gps disappear (gps working fine though)

Do you have a a donation paypal or such btw? ;)

**Comment by S4FOpel191190Michael** [comment]
Hallo das Problem mit dem VCA-Hud das es nicht angezeigt wird haben wir auch auf unseren Server!

Was kann das für ein Problem sein?

**Comment by Mogli12** [comment]
Please try build 13. I suspect that somehow the event data in client-sever-communication is corrupt. I added a begin and end marker to detect such problems.
Additionally, please delete vehicleControlAddon.xml in the savegame.

**Comment by Mogli12** [comment]
#620 -> German version

**Comment by c0nnex** [comment]
> Please try build 13. I suspect that somehow the event data in client-sever-communication is corrupt. I added a begin and end marker to detect such problems. Additionally, please delete vehicleControlAddon.xml in the savegame.

Build 16:
-  HUD is now Visible. Fixed
-  Bordermarks seem to stay on now (longer test following)
 
Settings:
- Stopped  Server , update mod,  Deleted vehicleControlAddon.xml, Restart Server , Enter Tractor, change VCA Settings,Exit Tractor, Force Game Save, leave Server , Stop Server:

Empty vehicleControlAddon.xml is created
`<?xml version="1.0" encoding="utf-8" standalone="no"?>
<vehicleControlAddon/>`
no file in local modsettings is created (that's what i would expect for per player settings)
Settings are persisted in vehcile.xml

- Restart Server , Join and check: VCA Settings all persisted.
(Could not check with 2 players yet, but since it's persited in vehicle.xml i assume it per Verhicle per Server not per Player)

When leaving server local log.txt logs
`2021-12-02 17:58 WARNING: object ref is invalid for resolving to script method target object
2021-12-02 17:58 LUA call stack:
`
(no call stack)
